src/validation.cpp

```diff
@@ -167,17 +166,19 @@ namespace {
     std::set<int> setDirtyFileInfo;
 } // anon namespace
 
-CBlockIndex* LookupBlockIndex(const uint256& hash)
+CBlockIndex* BlockManager::LookupBlockIndex(const uint256& hash)
 {
     AssertLockHeld(cs_main);
-    BlockMap::const_iterator it = g_chainman.BlockIndex().find(hash);
-    return it == g_chainman.BlockIndex().end() ? nullptr : it->second;
+    assert(std::addressof(g_chainman.BlockIndex()) == std::addressof(m_block_index));
+    BlockMap::const_iterator it = m_block_index.find(hash);
+    return it == m_block_index.end() ? nullptr : it->second;
 }
 
-CBlockIndex* FindForkInGlobalIndex(const CChain& chain, const CBlockLocator& locator)
+CBlockIndex* BlockManager::FindForkInGlobalIndex(const CChain& chain, const CBlockLocator& locator)
 {
     AssertLockHeld(cs_main);
 
+    assert(std::addressof(g_chainman.m_blockman) == std::addressof(*this));
```

Comment:
I think this `assert` can be removed after 98f2375, as `FindForkInGlobalIndex` is moved to `BlockManager` and thus you can now directly access the new local method `BlockManager::LookupBlockIndex`. Asserting `g_chainmain.m_blockman` equivalence is lapsed ? 

Same for `assert`s in `GetSpendHeight` after d34c069 and in `GetLastCheckpoint` after 866834a

Reply:
I see your point, but I don't think it hurts anything 😁
All of these will be removed in the end with a scripted-diff anyway